stencil/src/components/pulumi-api-doc-nav-tree/pulumi-api-doc-nav-tree.tsx

```diff
@@ -18,16 +24,44 @@ export class PulumiApiDocNavTree {
         );
     }
 
-    getChildNodes(nodes: node[] = this.nodes, depth: number = 0, linkBase = '') {
-        return nodes.map((node) => (
-            <pulumi-tree-item slot="item" expanded={!!node.isExpanded} class="nav-tree-item" title={node.name}>
-                {this.getIcon(node.type)}
-                <slot name="content">
-                    <a class={`depth-${depth}`} href={`${linkBase}${node.link}`}>{node.name}</a>
-                </slot>
-                {node.children && this.getChildNodes(node.children, depth + 1, linkBase + node.link)}
-            </pulumi-tree-item>
-        ));
+    // If a node is in the path for the page a user is currently on, it should
+    // be expanded, even if the expansion icon has not been clicked.
+    isNodeInPathForCurrentlyVisiblePage(nodeName) {
+        const currentPath = window.location.pathname;
+        return currentPath.includes(`/${nodeName}/`);
+    }
+
+    // If a node is an exact match for the page a user is currently on
+    // (not just in the path), it should be selected,
+    // even if it has not been clicked by the user.
+    shouldNodeBeSelected(nodePath) {
+        const currentPath = window.location.pathname;
+        return currentPath === `/${nodePath}/`;
+    }
+
+    getChildNodes(nodes: node[] = this.nodes, depth: number = 0, linkBase = "") {
+        return nodes.map((node) => {
+            const nodePath = `${this.docsDir}${linkBase}${node.link}`;
+            const nodeHref = `//${this.baseUrl}${nodePath}`;
```

Comment:
@cnunciato relative links will not work in this component, which is what i'm hoping to fix with this PR, happy to zoom and explain more about why that is.  For the time being, since local doesn't work with the lack of API docs, how do you feel about just leaving off the leading `//`?  That should work for staging + prod, and this isn't really a local dev-able thing anyway.